pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -383,7 +391,12 @@ func parseConfig(configReader io.Reader) (*Config, error) {
 }
 
 // Initialize passes a Kubernetes clientBuilder interface to the cloud provider
-func (az *Cloud) Initialize(clientBuilder controller.ControllerClientBuilder) {}
+func (az *Cloud) Initialize(clientBuilder controller.ControllerClientBuilder) {
+	az.kubeClient = clientBuilder.ClientOrDie("azure-cloud-provider")
```

Comment:
cc @tallclair @deads2k 
for more cloud provider clients

I know we don't want roles for these getting added to bootstrap policy. what client should they use to speak to the API?

Reply:
@liggitt Didn't get the response yet from @tallclair @deads2k. Any suggestion of candidate role should be used for such cases?